### PR TITLE
ci: merge staging to master

### PR DESCRIPTION
**Pull request by maxwell-aiden**
This is an automatic PR generated by the pipeline CI/CD. This will be automatically fast-forward merged if successful.

### CODE REVIEW

**Comment by maxwell-aiden** [comment]
Pipeline Attempt on 1181583170 for 88415e1e48c2f07c7182673d1948c89ff700fec8

https://gitlab.com/MatrixAI/open-source/Polykey-CLI/-/pipelines/1181583170

**Comment by maxwell-aiden** [comment]
Pipeline Attempt on 1181595214 for 082b6d96c30720acb55ebbcb2153180d300e7ba7

https://gitlab.com/MatrixAI/open-source/Polykey-CLI/-/pipelines/1181595214

**Comment by maxwell-aiden** [comment]
Pipeline Attempt on 1181669841 for 37e17277aacdbe77cdfe7fd4b0393426e17dc2dc

https://gitlab.com/MatrixAI/open-source/Polykey-CLI/-/pipelines/1181669841

**Comment by maxwell-aiden** [comment]
Pipeline Attempt on 1181704169 for 4b392f598d58d17ce3955f8a65ec4f41b0610b4b

https://gitlab.com/MatrixAI/open-source/Polykey-CLI/-/pipelines/1181704169

**Comment by maxwell-aiden** [comment]
Pipeline Attempt on 1181730992 for 1aa745900c7636e4017fae688e410d7546e0d077

https://gitlab.com/MatrixAI/open-source/Polykey-CLI/-/pipelines/1181730992

**Comment by maxwell-aiden** [comment]
Pipeline Succeeded on 1181730992 for 1aa745900c7636e4017fae688e410d7546e0d077

https://gitlab.com/MatrixAI/open-source/Polykey-CLI/-/pipelines/1181730992

File: flake.nix [hunk 1]
```diff
@@ -0,0 +1,161 @@
+{
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs?rev=ea5234e7073d5f44728c499192544a84244bf35a";
+    flake-utils.url = "github:numtide/flake-utils";
```
Comment (CMCDragonkai) [inline flake.nix:4]: Why is this needed? And is it something we would put into our `nixpkgs-matrix`?

Comment (CMCDragonkai) [reply flake.nix:4]: @brynblack

File: flake.nix [hunk 2]
```diff
@@ -0,0 +1,161 @@
+{
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs?rev=ea5234e7073d5f44728c499192544a84244bf35a";
```
Comment (CMCDragonkai) [inline flake.nix:3]: With the existence of `flake.lock`, what does it mean to pin it here?

Also next thing to do @brynblack is to replace this with `nixpkgs-matrix`!!
